win32ss/printing/base/winspool/printers.c

```diff
@@ -17,6 +17,92 @@
 static const WCHAR wszWindowsKey[] = L"Software\\Microsoft\\Windows NT\\CurrentVersion\\Windows";
 static const WCHAR wszDeviceValue[] = L"Device";
 
+BOOL strW2A(PWSTR wstring, PSTR pstring)
+{
+    // This converts an incoming Unicode string to an ANSI string
+    // Returns TRUE on failure, otherwise returns FALSE.
+
+    PSTR pszTemp;
+    DWORD cch;
+    BOOL bReturn = FALSE;
+
+    if (!wstring)
+    {
+        goto Exit;
+    }
+    else
+    {
+        cch = wcslen(wstring);
+    }
+
+    if (cch == 0)
+    {
+        goto Exit;
+    }
+
+    // Convert Unicode pName to a ANSI string pszTemp.
+    pszTemp = HeapAlloc(hProcessHeap, 0, (cch + 1) * sizeof(CHAR));
+    if (!pszTemp)
+    {
+        SetLastError(ERROR_NOT_ENOUGH_MEMORY);
+        ERR("HeapAlloc failed!\n");
+        bReturn = TRUE;   // indicates a failure to be handled by caller
+        goto Exit;
+    }
+
+    WideCharToMultiByte(CP_ACP, 0, wstring, -1, pszTemp, cch + 1, NULL, NULL);
+    StringCchCopyA(pstring, cch + 1, pszTemp);
+
+    HeapFree(hProcessHeap, 0, pszTemp);
+
+Exit:
+
+    return bReturn;
+}
+
+BOOL strA2W(PSTR pstring, PWSTR wstring)
```

Comment:
In-place conversions only work for Unicode to ANSI. It doesn't work the other way round, because the Unicode string always needs a larger buffer than the ANSI string.
I'm afraid you have to remove this function and do allocations and conversions manually for the ANSI to Unicode path. See for example `GetPrintProcessorDirectoryA`

Reply:
Fixed. Thanks.